src/autopas/selectors/tuningStrategy/PredictiveTuning.h

```diff
@@ -284,6 +319,9 @@ void PredictiveTuning::calculatePredictions() {
 
 void PredictiveTuning::linePrediction() {
   for (const auto &configuration : _searchSpace) {
+    if (_blacklistSearchSpace.count(configuration) > 0) {
+      continue;
+    }
```

Comment:
shouldn't this loop go over `_optimalSearchSpace`? (Same for the other prediction functions.)

Reply:
The _optimalSearchSpace got cleared with the prior reset and gets filled after the predictions got calculated.
_searchSpace is the only set that contains all configurations. There is no set, that only contains the not blacklisted functions.

Reply:
Then maybe there should be one ;)
I think it could make sense instead of having a set containing blacklisted configs to have a set with all not blacklisted configs. Then you could always loop over this and remove the checks whether a config is blacklisted.

Reply:
That would be easier in some way, but if we want so keep `_searchSpace`, which we should.
I have a few options:
- replace `_searchSpace` with `_whitelistSearchSpace` in the loops and `_whitelistSearchSpace` would contain `_searchSpace` if blacklisting was not used.
- create a iterationSearchSpacePointer, I would think of a better name, which would point to the searchSpace, that should be looped through with this it can be selected which should be used. Maybe useful for future functions.
- or something different I did not think of
But I would not want to select the searchSpace to loop through right before each loop.